prql/src/semantic/reporting.rs

```diff
@@ -0,0 +1,84 @@
+use std::ops::Range;
+
+use anyhow::{Ok, Result};
+use ariadne::{Color, Label, Report, ReportBuilder, ReportKind, Source};
+
+use super::{Declaration, SemanticAnalyzer, VarDec};
+use crate::ast::FuncDef;
+use crate::internals::{AstFold, Node};
+
+pub fn print(analyzer: &SemanticAnalyzer, source_id: String, source: String) {
```

Comment:
It looks like this isn't possible with the ariadne API, but generally a function that returns a `String` is easier to test and easier to build on. Atm it's difficult to use in a library.

(but obv fine for now)

Reply:
This is not intended to be used, it's only a tool for debugging. 

Ariadne also does not guarantee the layout will stay exactly the same, which means it would be hard to test.